flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java

```diff
@@ -745,20 +749,44 @@ public void visit(ListConstructorExpressionNode listCtrExprNode) {
     }
 
     private void handleConstructorExpressionNode(ExpressionNode constructorExprNode, String typeDescription) {
-        SyntaxKind kind = constructorExprNode.parent().kind();
+        NonTerminalNode parent = constructorExprNode.parent();
+        SyntaxKind kind = parent.kind();
         if (kind == SyntaxKind.LOCAL_VAR_DECL || kind == SyntaxKind.MODULE_VAR_DECL) {
+            Optional<Symbol> parentSymbol = semanticModel.symbol(parent);
+            if (parentSymbol.isPresent() && isVarTypeJson(parentSymbol.get())) {
+                startNode(FlowNode.Kind.JSON_PAYLOAD)
+                        .metadata()
+                        .description(JSONPayload.DESCRIPTION)
+                        .stepOut()
+                        .properties().expression(constructorExprNode);
+                return;
+            }
             startNode(FlowNode.Kind.NEW_DATA)
                     .metadata()
                         .description(NewData.DESCRIPTION, this.typedBindingPatternNode, typeDescription)
                         .stepOut()
                     .properties().expression(constructorExprNode);
         } else if (kind == SyntaxKind.ASSIGNMENT_STATEMENT) {
+            Optional<Symbol> parentSymbol = semanticModel.symbol(parent);
+            if (parentSymbol.isPresent() && isVarTypeJson(parentSymbol.get())) {
+                startNode(FlowNode.Kind.JSON_PAYLOAD)
+                        .metadata()
+                        .description(JSONPayload.DESCRIPTION)
+                        .stepOut()
+                        .properties().expression(constructorExprNode);
+                return;
+            }
             startNode(FlowNode.Kind.UPDATE_DATA).properties()
                     .expression(constructorExprNode)
-                    .variable(((AssignmentStatementNode) constructorExprNode.parent()).varRef());
+                    .variable(((AssignmentStatementNode) parent).varRef());
         }
     }
 
+    private static boolean isVarTypeJson(Symbol parentSymbol) {
+        return CommonUtils.getRawType(
+                ((VariableSymbol) parentSymbol).typeDescriptor()).typeKind() == TypeDescKind.JSON;
+    }
+
```

Comment:
Shall we move this to the `CommoUtils`?

Reply:
This function doesn't provide any generic functionality which is useful elsewhere. So I think better to keep it here.